racket/collects/racket/format.rkt

```diff
@@ -428,14 +462,34 @@
                  [else (let-values ([(q r) (quotient/remainder N base)])
                          (loop q (cons (get-digit r upper?) digits)))]))]))
 
+(define (number->string*/whole-positional N base upper? groups group-sep)
+  (if (zero? N)
+    (string #\0)    
+    (let loop ([N N] [digits null])
+      (cond
+        [(zero? N)
+         (apply string
+                (if (string=? group-sep "")
+                  digits
+                  (separate-groups (reverse digits)
+                                   (reverse groups)
+                                   (string->list group-sep))))]
+        [else
+         (let-values ([(q r) (quotient/remainder N base)])
+           (loop q (cons (get-digit r upper?) digits)))]))))
+
 (define (number->fraction-string N base upper? precision)
   (let ([s (number->string* N base upper?)])
     (string-append (make-string (max 0 (- precision (string-length s))) #\0) s)))
 
+(define A-10 (- (char->integer #\A) 10))
```

Comment:
Any reason why this part is changed? I find the old code less confusing -- `A-10` doesn't have a descriptive meaning by itself.

Reply:
This is just a little classic optimization. Printing numbers can be a little time sensitive, and the old code was (quite likely) doing some unnecessary work.